### DISCUSSION TITLE
Basic MIDI read function

### DISCUSSION THREAD

**Discussion by EdBeardon** [lathoub/Arduino-AppleMIDI-Library#143] [General]
Hi there, I appreciate this is pretty simple stuff but I would like to receive CC MIDI messages via RTP and use them in an ‘if’ statement to control LEDs. 

I’ve managed to send the messages to the arduino and they’re being shown on the serial monitor as ‘noteOn 60’ for example. I’m just wondering how I can interpret these messages into my code. 

I can’t seem to find any examples that explain this well.

Any help would be appreciated 
Ed

**Comment by Hackmancoltaire** [comment]
What I do in my code is have the noteOn and noteOff methods alter some global variables that my main loop goes through and sets things up. I'm using an old version of the framework but you can see an example here: https://github.com/Hackmancoltaire/mousepiano/blob/master/GarmentController/GarmentController.ino

**Reply by lathoub** [reply]
Thank you @Hackmancoltaire. Indeed, the question relates more to the underlying MIDI library than to the AppleMIDI transport layer. Your CC example should work just fine, even with the latest release, using the `MIDI.setHandleControlChange` callback. A more advanced CC example can be found [here](https://github.com/FortySevenEffects/arduino_midi_library/blob/master/examples/RPN_NRPN/RPN_NRPN.ino).
